xarray_beam/_src/core.py

```diff
@@ -330,22 +330,24 @@ def _first(self) -> xarray.Dataset:
 
   @property
   def _datasets(self) -> List[xarray.Dataset]:
-    if type(self.dataset) is xarray.Dataset:
+    if isinstance(self.dataset, xarray.Dataset):
       return [self.dataset]
     return list(self.dataset)
 
   def _validate(self, dataset, split_vars):
```

Comment:
Given how we support multiple input formats, it would be good to raise a `TypeError` for invalid inputs, i.e., anything other than `Dataset` or `list[Dataset]`. Otherwise somebody is likely to eventually try something like passing in a dict of Dataset objects -- and we should raise an error in that case.

Reply:
+1, I like that idea. I'll add that to the `_validate` method.